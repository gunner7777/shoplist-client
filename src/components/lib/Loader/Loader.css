.Loader {
  font-size: 45px;
  color: #26905c;
  height: 30px;
  width: 30px;
  animation: spinner 1s linear infinite; }

@keyframes spinner {
  0% {
    transform: rotate(0deg); }
  50% {
    transform: rotate(180deg); }
  100% {
    transform: rotate(360deg); } }
