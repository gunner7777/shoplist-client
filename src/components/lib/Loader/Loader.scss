@import '../../../style/colors.scss';

.Loader {
  font-size: 45px;
  color: $deep_blue;
  height: 30px;
  width: 30px;
  animation: spinner 1s linear infinite;
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(180deg);
  }
  100% {
    transform: rotate(360deg);
  }
}