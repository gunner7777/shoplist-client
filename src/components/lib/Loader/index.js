import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Loader.scss';

const Loader = () => (
  <div className="Loader-Outer Text_Center">
    <FontAwesomeIcon icon="spinner" className="Loader" />
  </div>
);

export default Loader;
