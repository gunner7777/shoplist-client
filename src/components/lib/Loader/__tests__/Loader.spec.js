import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../index';

describe('loader test', () => {
  it('loader snapshot', () => {
    const LoaderComponent = shallow(<Loader />);
    expect(LoaderComponent).toMatchSnapshot();
  });
});
