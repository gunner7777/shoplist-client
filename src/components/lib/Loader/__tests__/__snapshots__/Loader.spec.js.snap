// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader test loader snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FontAwesomeIcon
        border={false}
        className="Loader"
        fixedWidth={false}
        flip={null}
        icon="spinner"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />,
      "className": "Loader-Outer Text_Center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "border": false,
        "className": "Loader",
        "fixedWidth": false,
        "flip": null,
        "icon": "spinner",
        "inverse": false,
        "listItem": false,
        "mask": null,
        "pull": null,
        "pulse": false,
        "rotation": null,
        "size": null,
        "spin": false,
        "symbol": false,
        "title": "",
        "transform": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FontAwesomeIcon
          border={false}
          className="Loader"
          fixedWidth={false}
          flip={null}
          icon="spinner"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />,
        "className": "Loader-Outer Text_Center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "Loader",
          "fixedWidth": false,
          "flip": null,
          "icon": "spinner",
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
